process-engine/process_engine_runtime
# ⚡️✨Add lane and name to FlowNodeInstances

## Changes

1. Add `flowNodeLane` and `flowNodeName` to `FlowNodeInstanceModel`
2. Refactor ConsumerAPI and ManagmentAPI to use these values for their claim check operations
    - [Changelog ConsumerAPI](https://github.com/process-engine/consumer_api_core/pull/109)
    - [Changelog ManagementAPI](https://github.com/process-engine/management_api_core/pull/73)
3. Update all tests for both APIs
4. Add migration script for adding the new columns to the `FlowNodeInstance` database
5. Add post-migration script for updating existing records
6. Automatically run post-migration scripts during startup
7. Only set working directory after the post-migrations have run

## Issues

Closes #427

PR: #446

## How to test the changes

- Start up the ProcessEngineRuntime, using a database that does not yet have a `FlowNodeInstances` table with the columns `flowNodeName` and `flowNodeLane`
- See how the migrations will add the new columns
- Run the post-migration script `9.1.0-add_flow_node_instance_name_and_lane`
- Notice how all existing FlowNodeInstances that were run on a lane will get the name of the lane stored under `flowNodeLane`
- Also notice how each FlowNodeInstance will now have its name assigned to it
- Now make some queries through the ConsumerAPI and ManagementAPI for retrieving tasks
- In essence, these queries should work like before. However, if you query particularly large amounts of FlowNodeInstancs (like +1000), you should be able to notice an increase in performance; especially when querying EmptyActivities and ManualTasks
